python/ray/rllib/dqn/dqn.py

```diff
@@ -184,6 +207,69 @@ def get_gradient(self, cur_timestep):
             obses_t, actions, rewards, obses_tp1, dones = \
                 self.replay_buffer.sample(self.config["train_batch_size"])
             batch_idxes = None
+        replay_buffer_read_time = (time.time() - dt)
+        dt = time.time()
+        tuples_per_device = self.dqn_graph.multi_gpu_optimizer.load_data(
+            self.sess,
+            [obses_t, actions, rewards, obses_tp1, dones,
+             np.ones_like(rewards)])
+        per_device_batch_size = (
+            self.dqn_graph.multi_gpu_optimizer.per_device_batch_size)
+        num_batches = (int(tuples_per_device) // int(per_device_batch_size))
+        data_load_time = (time.time() - dt)
+        dt = time.time()
+        for _ in range(self.config["num_sgd_iter"]):
+            batches = list(range(num_batches))
+            random.shuffle(batches)
```

Comment:
why not `np.random.shuffle`, as `np` is already imported?

Reply:
Sure, I think it's the same.